iroha/src/smartcontracts/isi/permissions.rs

```diff
@@ -25,22 +25,30 @@ pub trait PermissionsValidator<W: WorldTrait> {
     fn check_instruction(
         &self,
         authority: &AccountId,
-        instruction: &Instruction,
+        operation: &O,
         wsv: &WorldStateView<W>,
     ) -> Result<(), DenialReason>;
 }
 
-/// Box with `PermissionChecker`
-pub type PermissionsValidatorBox<W> = Box<dyn PermissionsValidator<W> + Send + Sync>;
+/// Box with `PermissionsValidator`
+pub type PermissionsValidatorBox<W, O> = Box<dyn PermissionsValidator<W, O> + Send + Sync>;
+
+/// Box with `PermissionsValidator` for `Instruction`.
+pub type InstructionPermissionsValidatorBox<W> = PermissionsValidatorBox<W, Instruction>;
+
+/// Box with `PermissionsValidator` for `Query`.
+pub type QueryPermissionsValidatorBox<W> = PermissionsValidatorBox<W, QueryBox>;
```

Comment:
It became way too long. Can we rename it to smth like `InstructionPermitBox`, `QueryPermitBox` and `PermitBox`?

Reply:
I don't think it is too long :) And I want to stick with our rule to name only with full words

Reply:
4 words (28 symbols for `QueryPermissionsValidatorBox`!) for types is too long. Permit is a full word though...
QueryPermitBox though is 14 symbols while is full word

Reply:
Dont see any contradiction with our styleguide

Reply:
Permit is usually a legal term and it will not correspond with other names

Reply:
It is okay to have a long type - you rarely write it and even if you do you always have autocomplete

Reply:
I think we can change permission validator to some predicate: `IsAllowed`. I was thinking that permission validator is about validating permissions themselves, but it is actually about allowance to do stuff.

Reply:
I think as we discussed here the namings we should do a propper semantic naming and it should be not that long. I also think that as we have limit for lines of code per function (100 clippy default), we can also limit length of public items.

Reply:
Made oneliner for getting length of public items:
```sh
$ rg --no-heading 'pub (fn|struct|enum|mod|trait|const)' |
        sed -E \
            -e 's/.*pub (fn|struct|enum|mod|trait|const) //' \
            -e 's/[<(:;{].*//'|grep -v '[#$/]' | awk '{print length($1)}'  |
        sort |
        uniq -c |
        sort -h

   1 1
   1 40
   1 41
   2 36
   2 37
   2 42
   3 34
   3 35
   4 31
   6 28
   6 33
   8 29
   8 30
  10 27
  11 22
  11 23
  11 26
  12 25
  13 20
  13 24
  20 17
  29 19
  29 21
  30 18
  35 14
  36 9
  37 12
  37 16
  38 13
  41 15
  48 10
  55 11
  62 6
  63 5
  63 8
  80 2
  81 4
  89 7
 169 3
```

Reply:
I think that limiting it on 25 should be convinient. We would need to rename some of items, but generally i think codebase will benefit from reorganization and renaming of things. We dont use context aware namings in many places, so it would be easy to reorganize.

Reply:
But let it be soft-limit. It may be needed to use some longer name for something, if shorter name will not get it right.

Reply:
@i1i1 how do you think we should name PermissionValidatorBox, RecursivePermissionValidator, OrPermissionValidator ...
if we use IsAllowed trait?

Reply:
IsAllowedBoxed, Recur, Or, And, ...

Reply:
Ok, let's make an issue for this.

Reply:
I will probably take it in the next PR.

Reply:
https://jira.hyperledger.org/browse/IR-1154